### PR TITLE
For 5059: Add Glean metric for adding to autocomplete list from settings

### PR DESCRIPTION
**Pull request by mcarare**
Although there already is telemetry gathered for this, I wanted to ensure all newly requested telemetry can be gathered in the same place and also have proper data reviews. Also, this looked like low-hanging fruit to also maybe compare the 2 implementations' results for the same metric?!

### CODE REVIEW

**Comment by mcarare** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

_1) What questions will you answer with this data?_
How often do users add a site to the autocomplete list using the only remaining place, navigating from the settings screen.

_2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?_
Measure whether this feature is still used by the users after the removal of the option from the URL input screen.

_3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_
This is the only way.

_4) Can current instrumentation answer these questions?_
No. We need a new specific event for a specific app feature.

_5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._

_**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Add site to autocomplete list from settings</td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/focus-android/issues/5058</td>
  </tr>
</table>


_6) How long will this data be collected?  Choose one of the following:_
Until 2022-08-17.

_7) What populations will you measure?_
All release channels and locales.

_8) If this data collection is default on, what is the opt-out mechanism for users?_
 Users can opt of of data collection by disabling Usage and technical data from Settings -> Privacy and security -> Data choices.

_9) Please provide a general description of how you will analyze this data._
 Glean 

_10) Where do you intend to share the results of your analysis?_
Only on Glean, mobile teams and BD have internal access.

_12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_
No.

File: app/metrics.yaml [hunk 1]
```diff
@@ -130,3 +130,17 @@ settings_screen:
...
     expires: "2022-08-17"
+  autocomplete_domain_added:
+    type: counter
+    description: |
+      A counter that indicates how many times a user has added
+      a website to the autocomplete list.
+    bugs:
+      - https://github.com/mozilla-mobile/focus-android/issues/5059
+    data_reviews:
+      - https://github.com/mozilla-mobile/focus-android/pull/?
```
Comment (Amejia481) [inline app/metrics.yaml:141]: nit: Lets add the PR number

Comment (Amejia481) [reply app/metrics.yaml]: :)

**Review by rocketsroger — APPROVED** [review]

**Comment by mcarare** [comment]
@rocketsroger there is a problem with the UI tests and the task always fails. This however might be landed without ui test pass as it does not change UI or interactions. Still need a data review before that, tough. :)
```

Task :app:connectedFocusDebugAndroidTest
Starting 0 tests on test(AVD) - 11
Tests on test(AVD) - 11 failed: No test results

com.android.build.gradle.internal.testing.ConnectedDevice > No tests found.[test(AVD) - 11] FAILED 
No tests found. This usually means that your test classes are not in the form that your test runner expects (e.g. don't inherit from TestCase or lack @Test annotations).

 Task :app:connectedFocusDebugAndroidTest FAILED
```

I think  @AaronMT will look into this.

**Comment by rocketsroger** [comment]
> @rocketsroger there is a problem with the UI tests and the task always fails. This however might be landed without ui test pass as it does not change UI or interactions. Still need a data review before that, tough. :)

Yes, definitely still need the data-review to be approved.  👍

**Comment by travis79** [comment]
> # Request for data collection review form
> 
> **All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**
> 
> _1) What questions will you answer with this data?_
> How often do users add a site to the autocomplete list using the only remaining place, navigating from the settings screen.
> 
> _2) Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?_
> Measure whether this feature is still used by the users after the removal of the option from the URL input screen.
> 
> _3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_
> This is the only way.
> 
> _4) Can current instrumentation answer these questions?_
> No. We need a new specific event for a specific app feature.
> 
> _5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._
> 
> _**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Add site to autocomplete list from settings 	Category 2 	https://github.com/[/issues/5058](https://github.com/mozilla-mobile/focus-android/issues/5058)
> 
> _6) How long will this data be collected? Choose one of the following:_
> Until 2022-08-17.
> 
> _7) What populations will you measure?_
> All release channels and locales.
> 
> _8) If this data collection is default on, what is the opt-out mechanism for users?_
> Users can opt of of data collection by disabling Usage and technical data from Settings -> Privacy and security -> Data choices.
> 
> _9) Please provide a general description of how you will analyze this data._
> Glean
> 
> _10) Where do you intend to share the results of your analysis?_
> Only on Glean, mobile teams and BD have internal access.
> 
> _12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_
> No.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the Glean Dictionary

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the app settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end 2022-08-17

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Review by travis79 — APPROVED** [review]
data-review+

**Pull request merged**